<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>994</width>
    <height>669</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Random Walker Segmentation</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QVTKWidget" name="qvtkWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menu_bar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>994</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_save_viewpoint"/>
    <addaction name="action_load_viewpoint"/>
    <addaction name="separator"/>
    <addaction name="action_save_seeds"/>
    <addaction name="action_load_seeds"/>
    <addaction name="separator"/>
    <addaction name="action_save_segmentation"/>
    <addaction name="separator"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_view">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="action_graph_vertices"/>
    <addaction name="action_graph_edges"/>
    <addaction name="action_seeds"/>
    <addaction name="separator"/>
    <addaction name="action_highlight_cluster_mode"/>
   </widget>
   <widget class="QMenu" name="menu_segmentation">
    <property name="title">
     <string>Segmentation</string>
    </property>
    <addaction name="action_unique_labels"/>
   </widget>
   <addaction name="menu_file"/>
   <addaction name="menu_view"/>
   <addaction name="menu_segmentation"/>
  </widget>
  <widget class="QStatusBar" name="status_bar"/>
  <widget class="QDockWidget" name="dock_graph_builder">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>320</width>
     <height>212</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Graph builder</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dock_contents_graph_builder">
    <layout class="QVBoxLayout" name="verticalLayout_6">
     <item>
      <widget class="QTabWidget" name="tabs_graph_builder">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="currentIndex">
        <number>2</number>
       </property>
       <widget class="QWidget" name="tab_voxel_grid">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>279</width>
          <height>20</height>
         </size>
        </property>
        <attribute name="title">
         <string>Voxel Grid</string>
        </attribute>
        <layout class="QFormLayout" name="formLayout">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_voxel_resolution">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Voxel resolution</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="spinbox_voxel_resolution">
           <property name="decimals">
            <number>3</number>
           </property>
           <property name="minimum">
            <double>0.001000000000000</double>
           </property>
           <property name="maximum">
            <double>0.100000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.001000000000000</double>
           </property>
           <property name="value">
            <double>0.005000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_nearest_neighbors">
        <attribute name="title">
         <string>KNN</string>
        </attribute>
        <layout class="QFormLayout" name="formLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="label_nearest_neighbors">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Nearest neighbors</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="spinbox_nearest_neighbors">
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>30</number>
           </property>
           <property name="value">
            <number>15</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_radius">
        <attribute name="title">
         <string>Radius</string>
        </attribute>
        <layout class="QFormLayout" name="formLayout_2">
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
         </property>
         <item row="4" column="1">
          <widget class="QDoubleSpinBox" name="spinbox_radius">
           <property name="decimals">
            <number>3</number>
           </property>
           <property name="minimum">
            <double>0.001000000000000</double>
           </property>
           <property name="maximum">
            <double>10.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.001000000000000</double>
           </property>
           <property name="value">
            <double>0.010000000000000</double>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_radius">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Radius</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QSpinBox" name="spinbox_max_neighbors">
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>30</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_max_neighbors">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Max neighbors</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="button_update_graph">
       <property name="text">
        <string>Update</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dock_edge_weights">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>327</width>
     <height>177</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Edge weights</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dock_contents_edge_weights">
    <layout class="QVBoxLayout" name="verticalLayout_7">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QDoubleSpinBox" name="spinbox_xyz_influence">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>3.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkbox_xyz">
         <property name="text">
          <string>XYZ</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkbox_xyz_only_concave">
         <property name="text">
          <string>Only concave</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_9">
       <item>
        <widget class="QDoubleSpinBox" name="spinbox_normal_influence">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>0.005000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.005000000000000</double>
         </property>
         <property name="value">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkbox_normal">
         <property name="text">
          <string>Normal</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkbox_normal_only_concave">
         <property name="text">
          <string>Only concave</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_10">
       <item>
        <widget class="QDoubleSpinBox" name="spinbox_curvature_influence">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>0.000100000000000</double>
         </property>
         <property name="maximum">
          <double>0.100000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.000100000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkbox_curvature">
         <property name="text">
          <string>Curvature</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkbox_curvature_only_concave">
         <property name="text">
          <string>Only concave</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_11">
       <item>
        <widget class="QDoubleSpinBox" name="spinbox_rgb_influence">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>3.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkbox_rgb">
         <property name="text">
          <string>RGB</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkbox_rgb_only_concave">
         <property name="text">
          <string>Only concave</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dock_labels">
   <property name="features">
    <set>QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Labels</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dock_contents_labels">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QListView" name="list_labels">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>100</width>
         <height>0</height>
        </size>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::ExtendedSelection</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="button_new_label">
       <property name="text">
        <string>New label</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="button_delete_label">
       <property name="text">
        <string>Delete label</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="button_segment">
       <property name="text">
        <string>Segment</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="action_save_viewpoint">
   <property name="text">
    <string>Save viewpoint</string>
   </property>
  </action>
  <action name="action_load_viewpoint">
   <property name="text">
    <string>Load viewpoint</string>
   </property>
  </action>
  <action name="action_save_seeds">
   <property name="text">
    <string>Save seeds</string>
   </property>
  </action>
  <action name="action_load_seeds">
   <property name="text">
    <string>Load seeds…</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="action_graph_vertices">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Graph vertices</string>
   </property>
   <property name="shortcut">
    <string>V</string>
   </property>
  </action>
  <action name="action_graph_edges">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Graph edges</string>
   </property>
   <property name="shortcut">
    <string>E</string>
   </property>
  </action>
  <action name="action_save_segmentation">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save segmentation…</string>
   </property>
  </action>
  <action name="action_highlight_cluster_mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Highlight cluster mode</string>
   </property>
   <property name="shortcut">
    <string>H</string>
   </property>
  </action>
  <action name="action_seeds">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Seeds</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
  </action>
  <action name="action_unique_labels">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Unique labels for &quot;no-seeds&quot; clusters</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QVTKWidget</class>
   <extends>QWidget</extends>
   <header>QVTKWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>button_update_graph</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onButtonUpdateGraphClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>373</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_new_label</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>buttonNewLabelClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>506</y>
    </hint>
    <hint type="destinationlabel">
     <x>373</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_segment</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>buttonSegmentClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>506</y>
    </hint>
    <hint type="destinationlabel">
     <x>373</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_delete_label</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>buttonDeleteLabelClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>486</y>
    </hint>
    <hint type="destinationlabel">
     <x>373</x>
     <y>275</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_exit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionExitTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_load_viewpoint</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionLoadViewpointTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_save_viewpoint</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionSaveViewpointTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_load_seeds</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionLoadSeedsTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_save_seeds</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionSaveSeedsTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_graph_vertices</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionViewToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_graph_edges</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionViewToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_save_segmentation</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionSaveSegmentationTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_highlight_cluster_mode</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionViewToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_seeds</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionViewToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>buttonNewLabelClicked()</slot>
  <slot>buttonSegmentClicked()</slot>
  <slot>buttonDeleteLabelClicked()</slot>
  <slot>onActionExitTriggered()</slot>
  <slot>onActionLoadViewpointTriggered()</slot>
  <slot>onActionSaveViewpointTriggered()</slot>
  <slot>onActionLoadSeedsTriggered()</slot>
  <slot>onActionSaveSeedsTriggered()</slot>
  <slot>onButtonUpdateGraphClicked()</slot>
  <slot>onActionViewToggled(bool)</slot>
  <slot>onActionSaveSegmentationTriggered()</slot>
 </slots>
</ui>
